### DISCUSSION TITLE
add a way to control if a "run" statement runs after or before the rest of the migration

### DISCUSSION THREAD

**Discussion by trondhindenes** [piccolo-orm/piccolo#1336] [Ideas]
it's quite common to run "cleanup scripts" or "data migration scripts" after a migration's table updates - for example to populate some new table etc.

It would be good of piccolo expanded on the `manager.add_raw()` functionality so that it was possible to control if the custom function ran after or before the rest of the migration. As far as I can tell by default it runs before.

**Comment by dantownsend** [comment]
It's a good point. You're right - they always run before at the moment:

https://github.com/piccolo-orm/piccolo/blob/e1d5456a42927bb1f9a79781548f5c3b92010bee/piccolo/apps/migrations/auto/migration_manager.py#L716-L730

We could add something like `migration_manager.add_raw(some_function, called='after')`?

For now the workaround is to have a separate 'plain' migration which runs afterwards created using `piccolo migrations new my_app`. The downside though is it doesn't all run within the same transaction.

**Comment by dantownsend** [comment]
This behaviour definitely needs documenting - could catch people by surprise.

**Comment by trondhindenes** [comment]
yup we ended up with a separate migration too. `called="after"` would be nice! And yes, docs :-)
